bittensor/utils/async_substrate_interface.py

```diff
@@ -1704,6 +1705,24 @@ def make_payload(id_: str, method: str, params: list) -> dict:
             "payload": {"jsonrpc": "2.0", "method": method, "params": params},
         }
 
+    @lru_cache(maxsize=512)  # RPC methods are unlikely to change often
```

Comment:
`lru_cache` won't work here, as it will cache an awaitable, rather than the result. Either remove this or use something like [async-lru](https://github.com/aio-libs/async-lru).